lib/iex/lib/iex/autocomplete.ex

```diff
@@ -125,6 +128,51 @@ defmodule IEx.Autocomplete do
     end
   end
 
+  defp get_signatures(name, module) when is_atom(module) do
+    with docs when is_list(docs) <- get_docs(module, [:function, :macro], name) do
+      Enum.map(docs, fn {_, _, [signature], _, _} -> signature end)
+    else
+      _ -> :error
+    end
+  end
+
+  defp get_signatures(name, module, fs) when is_atom(module) and is_list(fs) do
+    fs
+    |> Enum.filter(&match?({^name, _arity}, &1))
+    |> Enum.map(fn {name, _} -> {module, name} end)
+    |> Enum.uniq()
+    |> Enum.map(fn {module, name} -> get_signatures(name, module) end)
+  end
```

Comment:
fs -> funs for readability.

Also, we can only have one matching name, so I would write this as:

```suggestion
  defp get_signatures(name, module, funs) when is_atom(module) and is_list(funs) do
    if List.keymember?(funs, name, 0) do
      get_signatures(name, module)
    else
      []
    end
  end
```